usmqe_tests/alerting/conftest.py

```diff
@@ -0,0 +1,171 @@
+
+    Returns:
+        dict: contains information about `start` and `stop` time of stress-ng
+            command and its `result`
+    """
+    def fill_cpu():
+        """
+        Use `stress-ng` tool to stress cpu for 3 minutes to given percentage
+        """
+        # stress cpu for for 180 seconds
+        run_time = 180
+        SSH = usmssh.get_ssh()
+        host = CONF.config["usmqe"]["cluster_member"]
+        processors_cmd = "grep -c ^processor /proc/cpuinfo"
+        retcode, processors_count, _ = SSH[host].run(processors_cmd)
+        stress_cmd = "stress-ng --cpu {} -l {} --timeout {}s".format(
+            int(processors_count),
+            fill_pct,
+            run_time)
+        retcode, stdout, stderr = SSH[host].run(stress_cmd)
+        if retcode != 0:
+            raise OSError(stderr)
+        return fill_pct
+    if request.param < 80:
+        fill_pct = 85
+        fill_cpu()
+    fill_pct = request.param
+    return measure_operation(fill_cpu)
+
+
+@pytest.fixture(params=[89, 30], scope="module")
+def workload_memory_utilization_alerts(request, stress_tools, alerting_setup):
+    """
+    Stress memory utilization on `cluster_member` node to value given in
+    parameter and measure the time it was utilized. If provided parameter
+    is below 80 (alert breach threshold) then the node is first utilized
+    to 89% and after that it is utilized again and measured.
+
+    Returns:
+        dict: contains information about `start` and `stop` time of stress-ng
+            command and its `result`
+    """
+    def fill_memory():
+        """
+        Use `stress` tool to stress memory for 3 minutes to given percentage
+        """
+        # stress memory for for 180 seconds
+        run_time = 180
+        SSH = usmssh.get_ssh()
+        host = CONF.config["usmqe"]["cluster_member"]
+        stress_cmd = "stress --vm-bytes $(awk '/MemAvailable/{{printf "\
+                     "\"%d\\n\" , $2 * ({0}/100);}}' < /proc/meminfo)k "\
+                     "--vm-keep -m {1}".format(fill_pct, 1)
+        stress_cmd += " --timeout {}s".format(
+            run_time)
+        retcode, stdout, stderr = SSH[host].run(stress_cmd)
+        if retcode != 0:
+            raise OSError(stderr)
+        return fill_pct
+    if request.param < 80:
+        fill_pct = 89
+        fill_memory()
+    fill_pct = request.param
+    return measure_operation(fill_memory)
+
+
+@pytest.fixture(scope="session")
+def alerting_setup(request):
+    """
+    Install and configure tools for alerting on client machine. Alerts will be
+    logged by:
+        SNMP
+        SMTP
+        Tendrl Api
+    """
+    with runner(
+            request,
+            [
+                "test_setup.alerts_logger.yml",
+                "test_setup.snmp.yml",
+                "test_setup.smtp.yml"],
+            [
+                "test_teardown.alerts_logger.yml",
+                "test_teardown.snmp.yml"]):
```

Comment:
I think we should run teardown for smtp as well after all. The teardown on the client doesn't uninstall mail clients nor deletes the emails.